src/utils/middleware/lollipop.ts

```diff
@@ -19,12 +21,28 @@ import {
   withLollipopHeadersFromRequest
 } from "../../types/lollipop";
 import { log } from "../logger";
+import { LollipopSignatureInput } from "../../../generated/lollipop/LollipopSignatureInput";
 
 type ErrorsResponses =
   | IResponseErrorInternal
   | IResponseErrorForbiddenNotAuthorized;
 
-export const lollipopMiddleware: (
+const LOLLIPOP_SIGN_ERROR_EVENT_NAME = "lollipop.error.sign";
+const NONCE_REGEX = new RegExp(';?nonce="(.+)";?');
```

Comment:
Is better to replace the regexp with
;?nonce="([^"]+)";?

The . will match anything, quote included. It's working becouse the current nodejs regex implementation will use the "first match" strategy, but this behaviour is empirical. I think it's safer to exclude quote from the matching group.

Reply:
You are right. The current regex is broken when there is another parameter after the `nonce`. For example:
```
sig-b21=();created=1618884473;nonce="b3k2pp5k7z-50gnwp.yemd";keyid="test-key-rsa-pss"
// group match with b3k2pp5k7z-50gnwp.yemd";keyid="test-key-rsa-pss
```
Exclude the `"` from the group make this character forbidden inside the nonce even backquoted but i think that we can introduce this limitation.